Add initial contribution guide (#1)
baikonur-oss/docs: CONTRIBUTING.md

# Contributing

> This doc explains how to get started contributing to [baikonur-oss](https://github.com/baikonur-oss) project.

Reviewer: IMHO official project name should be `Baikonur`. Thus, the link should be either `Baikonur` or `Baikonur OSS` (first preferred).

## Filing an Issue

> If you are trying to use baikonur-oss modules and run into some issues, please file an issue. We can look into the issue more easily if you describe in details as possible as you can. The following is example detailed descriptions:

Reviewer: Baikonur modules, for the same reason as above.

> - Version of tools and modules

Reviewer: PR templates to come?

Author: Yep. We should include a PR template in baikonur template repository. This is a temporary work.

- Expected and actual results
- Error messages with debug option
- Steps to reporoduce the issues

## Submitting a PR

> Before submitting a PR, make sure that there's an issue filed for your work. It might need some discussion before addressing the issue. Filing a issue first will help us to save your time to solving misunderstandings between you and us.

Reviewer: I doubt we need this section. 

Instead, I think we should say something like this:
```
PRs are welcome! Submitting an issue and discussing is a good practice, so feel free to open new issues. Baikonur project is young, and we still have to figure out guidelines etc., so submitting an issue prior to submitting a PR might be better.
```

## Releasing

> The release process for existing module goes something like this:

Reviewer: ```suggestion
The release process is as follows:
```

> 1. Make sure the direction for addressing the issues under discussion

Reviewer: I could not parse this. Please explain.

Author: This is related to the https://github.com/baikonur-oss/docs/pull/1#discussion_r305209300. I think we should discuss how to resolve the issue in filed Issue before accepting a PR.

Reviewer: I personally do not think we **must** have issues for PRs at all times.

Can we remove/comment out this line temporarily until we sort this out?

> 1. Accept a PR and review immpementation

Reviewer: ```suggestion
1. Accept a PR and review implementation
```

> 1. Additional E2E test or something using codes within the branch

Reviewer: ```suggestion
1. Additional E2E testing  (or unit tests etc. when available) with code on the PR branch. For Lambda modules, build packages from scratch and test.
```

1. Get approval and merge it into master

> 1. Increce version number on **master branch**

Reviewer: ```suggestion
1. Increase version number on **master branch**
```

Reviewer: Maybe even, `create a new incremented release from releases page`. 

Lambda packages also require a valid zip package attached with filename `lambda_package.zip`.

The release process for new module goes something like this:

1. Review first implementation including sample code pointing to the Terraform Registry URI
1. Put v0.0.1 tag on **PR branch**
1. Register module with Terraform Registry
1. Check the appearance of modules in Terraform Registry

> 1. Additional E2E test or something using codes tagged with v0.1.0 if needed

Reviewer: ```suggestion
1. Additional E2E, unit tests (with additional v0.0.x releases if needed)
```

1. Get approval and merge it into master

> 1. Put v1.0.0 tag on **master branch**

Reviewer: Not only tag, but we also want to have a [release](https://github.com/baikonur-oss/docs/releases).
